src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -1101,7 +1101,10 @@ template<class T> constexpr T MIN3(T a, T b, T c)      { return MIN2(MIN2(a, b),
 template<class T> constexpr T MAX4(T a, T b, T c, T d) { return MAX2(MAX3(a, b, c), d); }
 template<class T> constexpr T MIN4(T a, T b, T c, T d) { return MIN2(MIN3(a, b, c), d); }
 
-template<class T> inline T ABS(T x)                 { return (x > 0) ? x : -x; }
+template<class T> constexpr inline T ABS(T x) {
+  using U = std::make_unsigned_t<T>;
+  return (x >= 0) ? x : U(0) - U(x);
+}
```

Comment:
I understand what this to change to ABS is doing, though it's not obvious.  (Dodging overflow UB
for -x when x is the minimum value of a signed integral type.)  I'm not entirely sure that's a wise move.

As written this will trigger `-Wconversion` warnings someday (maybe).  static_casting the subtraction
result to T will eliminate that concern.

However, this is an API change.  The previous definition worked for floating point types, while this
change does not.  (std::make_unsigned requires T be an integral or enum, but not bool, type.)

I also don't understand why this change is part of this PR.

So I'm inclined to say no to this change without some compelling rationale.

Reply:
Sure I have reverted that also.